src/shared/components/person/reports.tsx

```diff
@@ -135,8 +144,38 @@ export class Reports extends Component<ReportsRouteProps, ReportsState> {
   }
 
   render() {
+    const removePostForm = this.state.removePostForm;
+    const removeCommentForm = this.state.removeCommentForm;
     return (
       <div className="person-reports container-lg">
+        {removePostForm && (
+          <ModActionFormModal
+            onSubmit={async (reason: string) => {
+              removePostForm.reason = reason;
+              await HttpService.client.removePost(removePostForm);
+              this.setState({ removePostForm: null });
+              this.update();
+            }}
+            modActionType="remove-post"
+            isRemoved={!removePostForm.removed}
+            onCancel={() => this.setState({ removePostForm: null })}
+            show={true}
+          />
+        )}
+        {removeCommentForm && (
+          <ModActionFormModal
+            onSubmit={async (reason: string) => {
+              removeCommentForm.reason = reason;
+              await HttpService.client.removeComment(removeCommentForm);
+              this.setState({ removeCommentForm: null });
+              this.update();
+            }}
+            modActionType="remove-comment"
+            isRemoved={!removeCommentForm.removed}
+            onCancel={() => this.setState({ removeCommentForm: null })}
+            show={true}
+          />
+        )}
```

Comment:
Is there any way to handle this generically, without a separate block for each type? Because it quickly gets very verbose. I tried to define a state `removeForm: RemovePost | RemoveComment` but there seems to be no good way to know the actual type and call the correct api endpoint.

Reply:
No, they're different forms anyway. Better to just be verbose and define specific handler functions for each.